components/dashboard/src/Menu.tsx

```diff
@@ -231,7 +231,10 @@ export default function Menu() {
                 link: `/t/${team.slug}/members`,
             },
         ];
-        if (showUsageView || (teamBillingMode && teamBillingMode.mode === "usage-based")) {
+        if (
+            currentUserInTeam?.role === "owner" &&
+            (showUsageView || (teamBillingMode && teamBillingMode.mode === "usage-based"))
+        ) {
```

Comment:
This would hide it from the UI, but leaves the APIs accessible. To actually restrict access, we need to change the APIs.

Would it make sense to continue showing it but show "You need to be a team Owner to access this view"? This helps with discovery, and action on what is needed to use it.

Reply:
> This would hide it from the UI, but leaves the APIs accessible. 

Oh, I didn't know that—thanks for the heads-up @easyCZ! 🏀 

> Would it make sense to continue showing it ...

We already show a message like this:

https://github.com/gitpod-io/gitpod/blob/40e78a254d2082085c09acded7cbce80799cca3f/components/dashboard/src/components/UsageView.tsx#L85

However, I see no benefit or much value for users to show this access wall, as there's not much they can do to access.

Cross-posting from the issue description in https://github.com/gitpod-io/gitpod/issues/11588:

> @gtsiolis Promoting a feature by making the usage tab visible and making the feature inaccessible not because of the user's subscription, but because of their team role can be confusing as it will probably be impossible for a user to change their role as this contradicts the initial goal of having roles in teams. This issue will become more obvious for larger teams which usually require specific role access so that they can control data access, and more.


Cross-postng from the [relevant discussion](https://gitpod.slack.com/archives/C01KPEPNBD0/p1671039700533679?thread_ts=1671022341.946149&cid=C01KPEPNBD0) (internal):

> @gtsiolis: Having this tab visible but not its content was a not-so-cool decision we took that we need to fix. I’ve also received negative feedback about this feature behavior in private conversations with other team members but we can fix this by either making this tab visible for all members or hiding the tab and improve usage attribution. :exclamation:

I'd suggest to :a: also limit API access to usage to team owners or :b: show usage to all team members. Happy to discuss more about this, or go with an MVC that requires the least engineering effort, but would be nice to have some product input or direction on this.

Reply:
I understood that we are already returning an error on the API but still show the UI. So hiding the UI seems to be the right thing to do here. In anyway I'd say we should merge this PR and do additional follow up work separately.

Reply:
Thanks @svenefftinge, wasn't aware there's already a guard for this. Makes sense to proceed with this!